src/org/gk/slicing/RevisionDetector.java

```diff
@@ -0,0 +1,442 @@
+		return events.stream()
+				.filter(attr -> attr.getSchemClass().isa(ReactomeJavaConstants.Pathway))
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Check if a ReactionlikeEvent is revised.
+	 *
+	 * @param reactionlikeEvent
+	 * @return boolean (true if revised, false otherwise).
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private boolean isRLERevised(GKInstance reactionlikeEvent, MySQLAdaptor targetDBA)
+			throws InvalidAttributeException, Exception {
+		// Check for changes in inputs, outputs, regulators, and catalysts.
+		Collection<String> revisionList = Arrays.asList(ReactomeJavaConstants.input,
+				ReactomeJavaConstants.output,
+				ReactomeJavaConstants.regulatedBy,
+				ReactomeJavaConstants.catalystActivity);
+
+		for (String attrName : revisionList) {
+			if (attributesRevised(reactionlikeEvent, getInstance(targetDBA, reactionlikeEvent), attrName))
+				return true;
+		}
+
+		// Check if a catalyst or regulator is added or removed.
+		Collection<String> additionsOrDeletionsList = Arrays.asList(ReactomeJavaConstants.regulatedBy,
+				ReactomeJavaConstants.catalystActivity);
+
+		for (String attrName : additionsOrDeletionsList) {
+			if (additionOrDeletionInList(reactionlikeEvent, getInstance(targetDBA, reactionlikeEvent), attrName))
+				return true;
+		}
+
+		// Check if summation is revised.
+		// Requires "special case" (examining "text" attribute for all summations).
+		if (isSummationRevised(reactionlikeEvent, targetDBA))
+			return true;
+
+		return false;
+	}
+
+	/**
+	 * Compare the value of a given attribute between two instances.
+	 *
+	 * Valid attribute checks take place in
+	 * {@link org.gk.slicing.SlicingEngine#isValidAttributeOrThrow(SchemaAttribute attribute)}
+	 *
+	 * @param oldInstance
+	 * @param newInstance
+	 * @param attrName
+	 * @return boolean (true if an attribute value is revised, false if all attribute values are equal).
+	 * @throws Exception
+	 */
+	private boolean attributesRevised(GKInstance oldInstance, GKInstance newInstance, String attrName)
+			throws Exception {
+		// Instance null checker.
+		if (oldInstance == null || newInstance == null) {
+			// if both instances are null, then there are no attributes to revise.
+			if (oldInstance == newInstance)
+				return false;
+			// if only one instance is null, then any attribute may be considered revised.
+			return true;
+		}
+
+		// Attribute null checker.
+		if (oldInstance.getAttributeValue(attrName) == null || newInstance.getAttributeValue(attrName) == null) {
+			if (oldInstance.getAttributeValue(attrName) == newInstance.getAttributeValue(attrName))
+				return false;
+			return true;
+		}
+
+		// If the object will not be able to cast to GKInstance, then simply compare the
+		// values.
+		if (oldInstance.getAttributeValue(attrName) instanceof String)
+			return !oldInstance.getAttributeValue(attrName).equals(newInstance.getAttributeValue(attrName));
```

Comment:
There are other types of attributes: Integer, Long, Boolean. How about these types of attribute values?

Reply:
Fixed in [cefec20](https://github.com/reactome/CuratorTool/pull/20/commits/cefec2021b7552cf14b97945dfc74a9b3727b3c2).